modules/021-cni-cilium/templates/safe-agent-updater/daemonset.yaml

```diff
@@ -0,0 +1,85 @@
+  namespace: d8-{{ $.Chart.Name }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "safe-agent-updater" "workload-resource-policy.deckhouse.io" "every-node")) | nindent 2 }}
+spec:
+  {{- include "helm_lib_resources_management_vpa_spec"  (list "apps/v1" "DaemonSet" "safe-agent-updater" "pause" .Values.cniCilium.resourcesManagement ) | nindent 2}}
+{{- end }}
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: safe-agent-updater
+  namespace: d8-{{ .Chart.Name }}
+  {{ include "helm_lib_module_labels" (list . (dict "app" "safe-agent-updater")) | nindent 2 }}
+spec:
+  selector:
+    matchLabels:
+      app: safe-agent-updater
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+  template:
+    metadata:
+      annotations:
+        configmap-checksum: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum | quote }}
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+      labels:
+        app: safe-agent-updater
+    spec:
+      {{- include "helm_lib_priority_class" (tuple . "system-node-critical") | nindent 6 }}
+      {{- include "helm_lib_tolerations" (tuple . "any-node" "with-uninitialized" "with-cloud-provider-uninitialized" "with-storage-problems") | nindent 6 }}
+      {{- include "helm_lib_module_pod_security_context_run_as_user_deckhouse" . | nindent 6 }}
+      terminationGracePeriodSeconds: 1
+      imagePullSecrets:
+      - name: deckhouse-registry
+      serviceAccountName: safe-agent-updater
+      initContainers:
+      {{- include "helm_lib_module_init_container_check_linux_kernel" (tuple . ">= 4.9.17") | nindent 6 }}
+      - name: prepull-image
+        {{- include "helm_lib_module_container_security_context_read_only_root_filesystem" . | nindent 8 }}
+        image: {{ include "helm_lib_module_image" (list . "cilium") }}
+        command:
+        - "/usr/bin/true"
+        terminationMessagePolicy: FallbackToLogsOnError
+        resources:
+          requests:
+            {{- include "helm_lib_module_ephemeral_storage_only_logs" . | nindent 12 }}
+      - name: safe-agent-updater
+        {{- include "helm_lib_module_container_security_context_read_only_root_filesystem" . | nindent 8 }}
+        image: {{ include "helm_lib_module_image" (list . "safeAgentUpdater") }}
+        imagePullPolicy: IfNotPresent
+        command:
+        - /safe-agent-updater
+        env:
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: spec.nodeName
+        resources:
+          requests:
+            {{- include "helm_lib_module_ephemeral_storage_only_logs" . | nindent 12 }}
+            {{- include "safe_agent_updater_resources" . | nindent 12 }}
+      containers:
+      - name: pause
+        {{- include "helm_lib_module_container_security_context_read_only_root_filesystem" . | nindent 8 }}
+        image: {{ include "helm_lib_module_common_image" (list . "pause") }}
```

Comment:
There must be cilium-agent image with pause binary like in [d8-control-plane-manager](https://github.com/deckhouse/deckhouse/blob/main/modules/040-control-plane-manager/images/kube-apiserver/werf.inc.yaml#L9).